cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,366 @@
+        )
+        parser.add_argument(
+            "--collection", type=str, help="Target iRODS collection. Skips manual selection input."
+        )
+        parser.add_argument(
+            "sources", help="One or multiple files/directories to ingest.", nargs="+"
+        )
+        parser.add_argument("destination", help="UUID or iRODS path of SODAR landing zone.")
+
+    @classmethod
+    def get_irods_error(cls, e: Exception):
+        """Return logger friendly iRODS exception."""
+        es = str(e)
+        return es if es != "None" else e.__class__.__name__
+
+    @classmethod
+    def run(
+        cls, args, _parser: argparse.ArgumentParser, _subparser: argparse.ArgumentParser
+    ) -> typing.Optional[int]:
+        """Entry point into the command."""
+        return cls(args).execute()
+
+    def execute(self):
+        """Execute ingest."""
+        # Get SODAR API info
+        toml_config = load_toml_config(Config())
+
+        if not self.args.sodar_url:
+            self.args.sodar_url = toml_config.get("global", {}).get("sodar_server_url")
+        if not self.args.sodar_api_token:
+            self.args.sodar_api_token = toml_config.get("global", {}).get("sodar_api_token")
+
+        # Retrieve iRODS path if destination is UUID
+        if is_uuid(self.args.destination):
+            try:
+                lz_info = api.landingzone.retrieve(
+                    sodar_url=self.args.sodar_url,
+                    sodar_api_token=self.args.sodar_api_token,
+                    landingzone_uuid=self.args.destination,
+                )
+            except Exception as e:
+                logger.error("Failed to retrieve landing zone information.")
+                logger.error(e)
+                sys.exit(1)
+
+            # TODO: Replace with status_locked check once implemented in sodar_cli
+            if lz_info.status in ["ACTIVE", "FAILED"]:
+                self.lz_irods_path = lz_info.irods_path
+                logger.info(f"Target iRods path: {self.lz_irods_path}")
+            else:
+                logger.info("Target landing zone is not ACTIVE.")
+                sys.exit(1)
+        else:
+            self.lz_irods_path = self.args.destination
+
+        # Build file list and add missing md5 files
+        source_paths = self.build_file_list()
+        source_paths = self.process_md5_sums(source_paths)
+
+        # Initiate iRODS session
+        irods_session = self._init_irods()
+
+        # Query target collection
+        logger.info("Querying landing zone collections…")
+        collections = []
+        try:
+            coll = irods_session.collections.get(self.lz_irods_path)
+            for c in coll.subcollections:
+                collections.append(c.name)
+        except:
+            logger.error("Failed to query landing zone collections.")
+            sys.exit(1)
+        finally:
+            irods_session.cleanup()
+
+        if self.args.collection is None:
```

Comment:
What if one wants to upload to multiple or all available collections?

Reply:
Again, I'm for keeping it simple and leave this kind of functionality to outside scripts or other, more specialised commands.